x/distribution/keeper/validator.go

```diff
@@ -91,6 +91,18 @@ func (k Keeper) GetValidatorAccum(ctx sdk.Context, operatorAddr sdk.ValAddress)
 	return accum, nil
 }
 
+// UpdateValidatorDistInfoFromPool updates the validator's distribution info
+// from the global fee pool without withdrawing any rewards. This will be called
+// from a onDelegationSharesModified hook on a non self bond/unbond call.
+func (k Keeper) UpdateValidatorDistInfoFromPool(ctx sdk.Context, operatorAddr sdk.ValAddress) {
+	valInfo := k.GetValidatorDistInfo(ctx, operatorAddr)
```

Comment:
this needs the error check which should likewise panic in the same way `onValidatorModified` panics
```
	if !k.HasValidatorDistInfo(ctx, operatorAddr) {
		return types.ErrNoValidatorDistInfo(k.codespace)
	}
```

Reply:
Updated!